include/OsmAndCore/Utilities.h

```diff
@@ -889,6 +889,38 @@ namespace OsmAnd
         static LatLon rhumbDestinationPoint(LatLon latLon, double distance, double bearing);
         static std::pair<int, int> calculateFinalXYFromBaseAndPrecisionXY(int bazeZoom, int finalZoom, int precisionXY,
                                                                           int xBase, int yBase, bool ignoreNotEnoughPrecision);
+        template<typename T>
+        inline static T computeDirection(
+            const T& start,
+            const T& end
+        )
+        {
+            auto length = std::sqrt((end.x - start.x)*(end.x - start.x) + (end.y - start.y)*(end.y - start.y));
+            return (end - start)/length;
+        }
+
+        template<typename T>
+        inline static T computeNormalToLine(
+            const T& start,
+            const T& end,
+            int dir
+        )
+        {
+            auto normal = computeDirection<T>(start, end);
+            T res;
+            if (0 > dir)
```

Comment:
what this comment means?

Reply:
That emoji means "I'm confused" as a way to express "I rarely see such way of comparing number and value, yet it's nothing wrong with it" :)

Reply:
Specifically here it's not important, but It's really make sense in case of comparison for instance - to avoid typo like:
`if (dir == 0)` and `if (dir = 0)`

Ref: https://en.wikipedia.org/wiki/Yoda_conditions

Reply:
You can't have that typo with `>` in any case, yet no hard objections there

Reply:
> You can't have that typo with...

Yes, exactly, as it mentioned before:
> Specifically here it's not important,

It's just a habit:)
Anyway it's already changed to the usual form.